### PR TITLE
Added graphical solution to example

### PR DESCRIPTION
**Pull request by lawrencefchan**
Added the fitted curve to demonstrate optimize-fit example.

Also added some lines to the plot code snippet to reflect output graph more closely.

### CODE REVIEW

File: intro/summary-exercises/optimize-fit.rst [hunk 1]
```diff
@@ -56,8 +56,14 @@ and visualize it::
...
     >>> t = np.arange(len(waveform_1))
-    >>> plt.plot(t, waveform_1) #doctest: +ELLIPSIS
+
+    >>> fig, ax = plt.subplots(figsize=(8, 6))
+    >>> ax.plot(t, waveform_1) #doctest: +ELLIPSIS
     [<matplotlib.lines.Line2D object at ...>]
+    >>> ax.set_xlabel('Time [ns]') #doctest: +ELLIPSIS
+    Text(0.5,0,'Time [ns]')
+    >>> ax.set_ylabel('Intensity [bins]') #doctest: +ELLIPSIS
+    Text(0,0.5,'Intensity [bins]')
```
Comment (lawrencefchan) [inline intro/summary-exercises/optimize-fit.rst]: I wasn't sure whether `#doctest ELLIPSIS ` should have been included, but my code didn't pass the tests without it.

**Comment by pdebuyl** [comment]
Dear @lawrencefchan 

Thank you for the contribution. I will review it later this week.

**Comment by pdebuyl** [comment]
Hello, here are my comments on the pull request:

1. There are existing graphics file, which we try to remove (see #262). For any update to the project, we do not take figure files but request that figures are generated: http://scipy-lectures.org/guide/index.html#figures-and-code-examples

    Could you transform your pull request to autogenerate the figures? Also, the third figure is superfluous.

2. The `+ELLIPSIS` option is only necessary when the code returns the object id, which is the case for plots and legends but not for the labels where they can be removed.

3. The usage of `subplot` and the related `ax.` method calls introduce unnecessary complexity. Please use the simpler `plt.plot`, `plt.xlabel`, etc.

I realize that these are big changes to your pull request, please let me know if you are interested to go on with the modifications.

Pierre

**Comment by lawrencefchan** [comment]
@pdebuyl thanks for the comments. I'll fix it and try again. Probably this weekend.

**Comment by lawrencefchan** [comment]
@pdebuyl sorry it's taken so long. how do the most recent changes look?

**Comment by pdebuyl** [comment]
I'll give it a look soon. Sorry for the delay. The merges end up with a lot of changes, many unrelated to your proposition. I'll have to sort it out.

**Comment by lawrencefchan** [comment]
Would it be easier if I closed this pull request and opened a new one from the current master?

**Comment by pdebuyl** [comment]
Yes, a pull request with only your changes would be easier.

Also:

1. The line "Solution: :ref:`Python source file <sphx_glr_intro_summary-exercises_auto_examples_plot_optimize_lidar_data.py>`" should be de-indented (else it is part of the "image" directive that takes no comment". Same for "Compare the result of :func:`scipy.optimize.leastsq`".

2. I'd like to keep the current narrative version, as in the rest of the lecture notes, where the code is show below its explanation. "load the waveform" followed by a few lines of Python code, and so on. There would be some duplication of code between the lecture page and the sphinx-gallery generated page from the Python file but that seems the best solution to me.

3. For many matplotlib commands, you don't need `#doctest: +ELLIPSIS`. This is only necessary for those that return a memory address ("line object at ..." for instance).

**Pull request closed without merging**